oxidation/libparsec/crates/serialization_format/src/utils.rs

```diff
@@ -4,23 +4,49 @@ use proc_macro2::TokenStream;
 use std::collections::HashMap;
 use syn::{GenericArgument, PathArguments, Type};
 
-pub(crate) fn snake_to_camel_case(s: &str) -> String {
-    let mut out = s[..1].to_uppercase();
-    let mut chars = s.chars().skip(1);
-    while let Some(c) = chars.next() {
-        if c == '_' {
-            match chars.next().unwrap_or_else(|| unreachable!()) {
-                c @ 'a'..='z' => out.push((c as u8 - b'a' + b'A') as char),
-                c => out.push(c),
+pub(crate) fn snake_to_pascal_case(s: &str) -> String {
+    let mut out = String::with_capacity(s.len());
+    // Start with capitalization
+    let mut capitalize_next_char = true;
+    for c in s.chars() {
+        match (capitalize_next_char, c) {
+            (_, '_') => {
+                capitalize_next_char = true;
+            }
+            (true, c @ 'a'..='z') => {
+                out.push((c as u8 - b'a' + b'A') as char);
```

Comment:
Use `take while` to consume the next chars that are in lowercase ?

Reply:
What do you mean ? We already iterate over `s.chars()`, how would you jump after ?

Reply:
I mean this in this case (and the others cases) we would consume the next chars that are alike (lowercase char, spaces, etc)

Reply:
But `take_while`, consume the `iterator` so your idea will generate many `iterators`

Reply:
You're right, so my suggestion isn't possible :shrug: